### PR TITLE
Upstream FreeBSD patches

### PR DESCRIPTION
**Pull request by abishai**
Now ZM compiles on FreeBSD without any patches. Not sure if it still runs, I see a lot of new stuff was merged to the master.

Observation: cmake looks for boost, but I found no boost usage in the code.

### CODE REVIEW

**Comment by knnniggett** [comment]
Thanks for this. 
See #1902 and #1908 for several new cmake variables you must set at build time.

**Comment by knnniggett** [comment]
This pr looks fine by me

File: scripts/CMakeLists.txt [hunk 1]
```diff
@@ -33,8 +33,11 @@ FOREACH(PERLSCRIPT ${perlscripts})
...
+install(FILES "${CMAKE_CURRENT_BINARY_DIR}/zmaudit.pl" "${CMAKE_CURRENT_BINARY_DIR}/zmcontrol.pl" "${CMAKE_CURRENT_BINARY_DIR}/zmdc.pl" "${CMAKE_CURRENT_BINARY_DIR}/zmfilter.pl" "${CMAKE_CURRENT_BINARY_DIR}/zmpkg.pl" "${CMAKE_CURRENT_BINARY_DIR}/zmtrack.pl" "${CMAKE_CURRENT_BINARY_DIR}/zmtrigger.pl" "${CMAKE_CURRENT_BINARY_DIR}/zmupdate.pl" "${CMAKE_CURRENT_BINARY_DIR}/zmvideo.pl" "${CMAKE_CURRENT_BINARY_DIR}/zmwatch.pl" "${CMAKE_CURRENT_BINARY_DIR}/zmcamtool.pl" "${CMAKE_CURRENT_BINARY_DIR}/zmtelemetry.pl" DESTINATION "${CMAKE_INSTALL_FULL_BINDIR}" PERMISSIONS OWNER_WRITE OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
 if(NOT ZM_NO_X10)
 	install(FILES "${CMAKE_CURRENT_BINARY_DIR}/zmx10.pl" DESTINATION "${CMAKE_INSTALL_FULL_BINDIR}" PERMISSIONS OWNER_WRITE OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
 endif(NOT ZM_NO_X10)
 
+if(WITH_SYSTEMD)
+    install(FILES "${CMAKE_CURRENT_BINARY_DIR}/zmsystemctl.pl" PERMISSIONS OWNER_WRITE OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE WORLD_READ WORLD_EXECUTE)
```
Comment (knight-of-ni) [inline scripts/CMakeLists.txt]: @abishai Travis CI failed the build because this line is missing `DESTINATION "${CMAKE_INSTALL_FULL_BINDIR}"`

**Comment by abishai** [comment]
Also, I have BSD init script and readme. Should I put it in a folder under distro  ?

File: scripts/CMakeLists.txt (hunk 1 above)
Comment (abishai) [reply scripts/CMakeLists.txt]: Fixed. Thanks!

**Comment by knnniggett** [comment]
Yes, you can certainly create a distros/freebsd subfolder. Feel free to create a CMakeLists.txt in that folder as well if you want cmake to install those files onto the target filesystem. Let me know if you need help doing that.

**Comment by abishai** [comment]
Sure. I will, when I'll be ready to dig into cmake magic.
Thanks for merge.

**Pull request merged**